File: frontend/src/features/ai-review/types/aiReview.ts
// api-client가 응답을 자동 camelize 하므로 모든 필드는 camelCase.
export interface DailyReview {
  babyId: string;
  reviewDate: string;
  feedingAnalysis: string;
  sleepAnalysis: string;
  diaperAnalysis: string;
  playAnalysis: string;
  overallAssessment: string;
  alerts: string[];
  recommendations: string[];
  positives: string[];
  considerations: string[];
  concerns: string[];
  criticalWarnings: string[];
}

export interface SavedInfo {
  id: string;
  babyId: string;
  title: string;
  content: string;
  category: "feeding" | "sleep" | "development" | "health" | "general";
  chatMessageId?: string;
  createdAt: string;
}

File: backend/app/domain/entities/ai_review.py
from dataclasses import dataclass, field
from datetime import date, datetime
from uuid import UUID


@dataclass
class AIReview:
    id: UUID
    baby_id: UUID
    review_date: date
    feeding_analysis: str
    sleep_analysis: str
    diaper_analysis: str
    play_analysis: str
    overall_assessment: str
    alerts: list[str]
    recommendations: list[str]
    created_at: datetime
